### PR TITLE
add variadic mode and median functions

### PR DESCRIPTION
**Pull request by jmbothe**
addresses issues #27 and #28

### CODE REVIEW

File: lib/mode.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const num = acc.find(obj => obj.value === cur);
+    num ? num.frequency += 1 : acc.push({ value: cur, frequency: 1 });
+    return acc;
+  }, []);
+
+  // if there is only one number in the list, return its value and frequency
+  if (valFreq.length === 1) return valFreq;
+
+  // if all frequencies are the same, the set has no mode
+  if (valFreq.every(num => num.frequency === valFreq[0].frequency)) return 'set has no mode';
```
Comment (jhowardjr) [inline lib/mode.js]: Probably could return false here.

Comment (jmbothe) [reply lib/mode.js]: ah yes. good call @jhowardjr . actually, how about `undefined`? That would probably make the most sense.

Comment (jhowardjr) [reply lib/mode.js]: ```undefined``` is good too.

Comment (jmbothe) [reply lib/mode.js]: now that you've got me thinking about output, I've started considering re-structuring the return value of a hit to something like below. what do you think?

```
// input: 1, 1, 1, 1, 2, 2, 2, 2, 3

// output:
{
  mode: [1, 2],
  frequency: 4,
}
```

seems more consice than:
```
[
  { value: 1, frequency: 4 },
  { value: 2, frequency: 4 }
]
```

After all, the frequency is going to be the same for all values...

Comment (jhowardjr) [reply lib/mode.js]: How would the frequency be used outside of determining the mode within the function? A few principles I generally follow in practice you aren't going to need it (YAGNI) and KISS. Frequency could probably be its own function if there were a use outside of the mode function.

Comment (jmbothe) [reply lib/mode.js]: Good point. I'll go ahead and make it so `mode()` simply returns an array of the set's mode(s), then push another commit to this PR. I'll think about use cases for a separate `frequency()` function, and if it warrants it, I can always factor it out and make a different PR. sound good?

Comment (jhowardjr) [reply lib/mode.js]: software is always changing and is never done!

**Pull request merged**